codec-http/src/main/java/io/netty/handler/codec/http/HttpHeadersFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+/**
+ * An interface for creating {@link HttpHeaders} instances.
+ * <p>
+ * The default implementation is {@link HttpHeadersBuilder},
+ * and the default instance is {@link HttpHeaders#DEFAULT_HEADER_FACTORY}.
+ */
+public interface HttpHeadersFactory {
+    /**
+     * Create a new {@link HttpHeaders} instance.
+     */
+    HttpHeaders createHeaders();
```

Comment:
For similar use-cases, I tend to avoid having an interface with only one method bcz it automatically becomes functional. It makes it harder in the future to expand such interface, assuming that there might be a point when it's ok to drop `default` implementations accepting incompatibility, but lambdas make all users experience the worst in such cases.

Consider adding another method, like `createEmptyHeaders()`. Sometimes it's useful to grab a headers object just to satisfy API without actually modifying them. For example, for trailers. 

Also, not that it's required, but might be nice to keep naming consistent with `HttpHeadersFactory` in ServiceTalk, it uses `newHeaders()` and `newEmptyHeaders()` insated of "create".